### PR TITLE
Closes #2513 accept_ms: accept ms as input unit for derive_param_qtc()

### LINKED ISSUE #2513: Bug: derive_param_qtc() expects QT and RR in "msec"
### What happened?

In fact this is not really a bug but a sort of obsolescence. It was fine at the time when `derive_param_qtc()` was created. However, on 2022-06-24 the CDISC Submission Value for milliseconds was changed from `msec` to `ms` in the SDTM Controlled Terminology.

See line 1084 in [SDTM Terminology Changes 2022-06-24.xls](https://evs.nci.nih.gov/ftp1/CDISC/SDTM/Archive/SDTM%20Terminology%20Changes%202022-06-24.xls)

So in all clinical trials started after that date, the following error is thrown:

```
Error in `derive_param_qtc()`:
! It is expected that "QT" has unit of "msec". In the input dataset the unit is "ms".
Run `rlang::last_trace()` to see where the error occurred.
```

### Session Information

_No response_

### Reproducible Example

```R
library(tibble)

adeg <- tribble(
  ~USUBJID, ~PARAMCD, ~PARAM, ~AVAL, ~AVALU, ~VISIT,
  "01-701-1015", "HR", "Heart Rate (beats/min)", 70.14, "beats/min", "SCREENING",
  "01-701-1015", "HR", "Heart Rate (beats/min)", 62.66, "beats/min", "WEEK 1",
  "01-701-1015", "RR", "RR Duration (ms)",       710,   "ms",        "WEEK 2",
  "01-701-1015", "QT", "QT Duration (ms)",       370,   "ms",        "WEEK 2",
  "01-701-1028", "HR", "Heart Rate (beats/min)", 85.45, "beats/min", "SCREENING",
  "01-701-1028", "HR", "Heart Rate (beats/min)", 56.54, "beats/min", "WEEK 3",
  "01-701-1028", "RR", "RR Duration (ms)",       842,   "ms",        "WEEK 2",
  "01-701-1028", "QT", "QT Duration (ms)",       480,   "ms",        "WEEK 2",
  "01-701-1028", "QT", "QT Duration (ms)",       350,   "ms",        "WEEK 3"
)

derive_param_qtc(
  adeg,
  by_vars = exprs(USUBJID, VISIT),
  method = "Fridericia",
  set_values_to = exprs(
    PARAMCD = "QTCFR",
    PARAM = "QTcF - Fridericia's Correction Formula Rederived (ms)",
    AVALU = "ms"
  ),
  get_unit_expr = extract_unit(PARAM)
)
```

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bundfussr** [comment]
The failing R-CMD checks will pass once https://github.com/pharmaverse/admiraldev/issues/468 is merged.

File: R/derive_adeg_params.R [hunk 1]
```diff
@@ -24,59 +24,62 @@
...
+#'   See [compute_qtc()] for details.
+#'
+#'   *Permitted Values*: `"Bazett"`, `"Fridericia"`, `"Sagie"`
 #'
 #' @param qt_code QT parameter code
 #'
 #'   The observations where `PARAMCD` equals the specified value are considered
-#'   as the QT interval assessments. It is expected that QT is measured in msec.
+#'   as the QT interval assessments. It is expected that QT is measured in msec
+#'   or ms.
```
Comment (yurovska) [inline R/derive_adeg_params.R:35]:
```suggestion
#'   as the QT interval assessments. It is expected that QT is measured in ms
#'   or msec.
```

File: R/derive_adeg_params.R [hunk 2]
```diff
@@ -24,59 +24,62 @@
...
 #'
-#'   Permitted Values: character value
+#'   *Permitted Values*: character value
 #'
 #' @param rr_code RR parameter code
 #'
 #'   The observations where `PARAMCD` equals the specified value are considered
-#'   as the RR interval assessments. It is expected that RR is measured in msec.
+#'   as the RR interval assessments. It is expected that RR is measured in msec
+#'   or ms.
```
Comment (yurovska) [inline R/derive_adeg_params.R:43]:
```suggestion
#'   as the RR interval assessments. It is expected that RR is measured in ms
#'   or msec.
```

File: R/derive_adeg_params.R [hunk 3]
```diff
@@ -196,16 +198,19 @@ default_qtc_paramcd <- function(method) {
 #'
 #' @param qt QT interval
 #'
-#'   A numeric vector is expected. It is expected that QT is measured in msec.
+#'   A numeric vector is expected. It is expected that QT is measured in msec or
+#'   ms.
```
Comment (yurovska) [inline R/derive_adeg_params.R:202]:
```suggestion
#'   A numeric vector is expected. It is expected that QT is measured in ms or
#'   msec.
```

File: R/derive_adeg_params.R [hunk 4]
```diff
@@ -196,16 +198,19 @@ default_qtc_paramcd <- function(method) {
...
 #'
 #' @param rr RR interval
 #'
-#'   A numeric vector is expected. It is expected that RR is measured in msec.
+#'   A numeric vector is expected. It is expected that RR is measured in msec or
+#'   ms.
```
Comment (yurovska) [inline R/derive_adeg_params.R:207]:
```suggestion
#'   A numeric vector is expected. It is expected that RR is measured in ms or
#'   msec.
```

File: NEWS.md [hunk 5]
```diff
@@ -12,6 +12,9 @@
 
 ## Updates of Existing Functions
 
+- The `derive_param_qtc()` function accepts now both `"msec"` and `"ms"` as unit
```
Comment (yurovska) [inline NEWS.md:15]:
```suggestion
- The `derive_param_qtc()` function accepts now both `"ms"` and `"msec"` as unit
```

**Review by yurovska — APPROVED** [review]
Just a super minor comment regarding the order of "ms" and "msec". I think "ms" should go first everywhere in the documentation as this is the standard unit now. Otherwise I'm happy with the changes.

File: tests/testthat/test-derive_adeg_params.R
Comment (bms63) [inline tests/testthat/test-derive_adeg_params.R:1]: Should we have a test for differing units?  or is that already covered and just not seeing it??

**Review by bms63 — COMMENTED** [review]
@bundfussr what do you think about splitting out the functions into separate files?  having everything in `ad_adeg.R` has always irked me a bit.  

We can make this a good first issue and have someone new do this and not a priority

...we only do this type of organization for this and the advs_params.

File: R/derive_adeg_params.R (hunk 1 above)
Comment (bundfussr) [reply R/derive_adeg_params.R]: Updated

File: R/derive_adeg_params.R (hunk 2 above)
Comment (bundfussr) [reply R/derive_adeg_params.R]: Updated

File: R/derive_adeg_params.R (hunk 3 above)
Comment (bundfussr) [reply R/derive_adeg_params.R]: Updated

File: R/derive_adeg_params.R (hunk 4 above)
Comment (bundfussr) [reply R/derive_adeg_params.R]: Updated

File: NEWS.md (hunk 5 above)
Comment (bundfussr) [reply NEWS.md]: Updated

**Comment by bundfussr** [comment]
> @bundfussr what do you think about splitting out the functions into separate files? having everything in `ad_adeg.R` has always irked me a bit.
> 
> We can make this a good first issue and have someone new do this and not a priority
> 
> ...we only do this type of organization for this and the advs_params.

Yes, we could do this but I would split by endpoint and not by function. For example, I would keep `derive_param_qtc()` and `compute_qtc()` in the same file.

By the way, I would like to split `assertions.R` and `test-assertions.R` in admiraldev. They have well over 1000 lines. Should I make a proposal how to split them for the next developer meeting?

File: tests/testthat/test-derive_adeg_params.R
Comment (bms63) [reply tests/testthat/test-derive_adeg_params.R:1]: sorry i meant the tests units are ms and msec in the same the dataset.  i have seen this happen a few times over the years where units from different sites differ and are not caught until way later.

**Comment by bms63** [comment]
> > @bundfussr what do you think about splitting out the functions into separate files? having everything in `ad_adeg.R` has always irked me a bit.
> > We can make this a good first issue and have someone new do this and not a priority
> > ...we only do this type of organization for this and the advs_params.
> 
> Yes, we could do this but I would split by endpoint and not by function. For example, I would keep `derive_param_qtc()` and `compute_qtc()` in the same file.
> 
> By the way, I would like to split `assertions.R` and `test-assertions.R` in admiraldev. They have well over 1000 lines. Should I make a proposal how to split them for the next developer meeting?

YES PLEASE!!  LOVE IT ALL!!!!!!

File: tests/testthat/test-derive_adeg_params.R
Comment (bundfussr) [reply tests/testthat/test-derive_adeg_params.R:1]: The unit must be unique within a parameter. This is also checked by `assert_unit()`. We test it in the unit tests for `assert_unit()`. I'm not sure if we need to test it for `derive_param_qtc()`. Then we would need to test other bad inputs like a wrong unit as well. Usually we don't do this.

Comment (bms63) [reply tests/testthat/test-derive_adeg_params.R:1]: If you feel confident that this would be caught, then all is good for me.

**Comment by bms63** [comment]
@bundfussr I feel like this is good to merge in - the step you add in the GHA makes sense and takes out a manual step.  We can always revert if for some reason Remotes has to get back in.
